File: src/Messages/TablesAndShips.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Messages.CSharp
{
    public static class TablesAndShips
    {
        
        public static readonly IReadOnlyCollection<Tuple<string, int>> Ships
            = new ReadOnlyCollection<Tuple<string, int>>(new List<Tuple<string, int>>
            {
                Tuple.Create("Carrier", 5),
                Tuple.Create("BattleShip", 4),
                Tuple.Create("Cruiser", 3),
                Tuple.Create("Cruiser", 3),
                Tuple.Create("Destroyer", 2),
                Tuple.Create("Submarine", 1)
            });


    }
}

File: src/Messages.FSharp/TablesAndShips.fs
module Messages.FSharp.TablesAndShips

    type Piece = {
        Name:string
        Length:int}

    let private pieces = [
        { Name = "Carrier"; Length = 5 }
        { Name = "BattleShip"; Length = 4 }
        { Name = "Cruiser"; Length = 3 }
        { Name = "Cruiser"; Length = 3 }
        { Name = "Destroyer"; Length = 2 }
        { Name = "Submarine"; Length = 1 }
    ]
    
    let GetTablesAndShips() = pieces
